### PR TITLE
🥳[Feature] Transfer from Litewallet to Litecoin Card

### LINKED ISSUE #73: 🥳[Feature] Add Transfer Capability between Litewallet and Litecoin Card
## Overview
Final phase for the Litecoin card integration this should allow the user to transfer from to or from Litewallet or Litecoin Card
### Why?
Allow users to transfer Litecoin between either

## Action
|Litewallet Balance|Litecoin Card Balance| Action|
|---|----|---|
| 0Ł | >0Ł| Card moves to allow deposit to Litewallet|
| >0Ł | >0Ł| Either Litewallet icon or Litecoin Card can move to allow deposit|
| 0Ł | 0Ł | No movement or either|
| >0Ł | 0Ł| Litewallet moves to allow deposit to Litecoin Card|

## Draft v1
LitecoinCardTransfer-v1

 
|**Before**|**After**|
|----------|---------|
|||

### PR DESCRIPTION
**Pull request by kcw-grunt**
## Overview
The main integration of the Litecoin Card is to allow the user to transfer Litecoin between the Litecoin Card and Litewallet. This work will allow LTC Transfer:
`Litewallet` -> ` Litecoin Card`

## Approach
This work was always expected to be very comprehensive and complex as there are a few preliminary steps needed to be completed in order for the process to work.

1. UI / UX Design (_8 - 13 Points_)
2. Business Logic to calculate the amounts, and set the metadata (_5  - 8 Points_)
3. Implement and refactor the API Calls in the LitewalletPartner API Library (_3 - 5 Points_)
4. Implement the API Calls in the iOS Client (_3  - 5 Points_)
5. Write Tests (_3  - 5 Points_)

Most of the valuation of each step was misbid as the work took **~4 weeks**

## Sub-tasks and GIFs
Note some of the sub-tasks for completing the steps

| Enable 2FA GIF | The PreTransfer View | The Amount View | 
|----------------|---------------------|----------------------|
|2FA_480|PreTransferView|AmountView|

## Resolution and Completion
 The goal was to have full 2-way transfer and the problem for the Ternio API is the documentation is unclear or contradictory.
A new issue to address the `Litecoin Card` -> `Litewallet`  Transfer capability is has been written: 
This will be included once the process for the withdrawal is clarified from Ternio. 

The second and final step of the Card integration is to have `Litecoin Card` -> `Litewallet` transfer.   An issue is published here: #85 

## Closes
#73

### CODE REVIEW

File: loafwallet/Enter2FACodeView.swift [hunk 1]
```diff
@@ -0,0 +1,154 @@
+//
+//  Enter2FACodeView.swift
+//  loafwallet
+//
+//  Created by Kerry Washington on 8/9/21.
+//  Copyright © 2021 Litecoin Foundation. All rights reserved.
+//
+
```
Comment (kcw-grunt) [inline loafwallet/Enter2FACodeView.swift:8]: Reusing this architecture for any modal view. The idea is presenting a customized modal view is a nice option as compared to Action Sheets or Alerts.  This view is presented in the View+Extension class.

File: loafwallet/Enter2FACodeView.swift [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                            .padding(.bottom, 8)
+                            .foregroundColor(Color.white)
+                         
+                        Text("XXX Enter the code that was sent to your Litecoin Card account email.")
+                            .font(Font(UIFont.barlowRegular(size: 19.0)))
+                            .foregroundColor(Color.white)
+                            .padding([.leading, .trailing], 20)
+
+                        
+                        TextField("", text: $twoFAViewModel.tokenString)
```
Comment (kcw-grunt) [inline loafwallet/Enter2FACodeView.swift:61]: This is a good example of how SwiftUI works with a MVVM architecture Here the `twoFAViewModel.tokenString` is updated from a `didSet`where it also filtered the length and char type

File: loafwallet/Enter2FACodeView.swift [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                            .background(Color.white)
+                        
+                        HStack {
+                            
+                            // Confirm button
+                            Button(action: {
+                            
+                                
+                                twoFAViewModel.didConfirmToken { token in
+                                    twoFAViewModel.tokenString = token
```
Comment (kcw-grunt) [inline loafwallet/Enter2FACodeView.swift:88]: Also tests the code is of the correct length

**Comment by kcw-grunt** [comment]
Let's meet to go over this. Again, sorry about the length  @JasperHeist

File: loafwallet/TransferAmountViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            .shared
+            .withdrawToWallet(userID: userID,
+                              token: token,
+                              withdrawal:
+                                ["amount": amount,
+                                 "wallet_address": address]) { dict in
+                
+               // self.keychain["2fatoken"] = self.tokenString
+
+                // Successful
```
Comment (JCHeist) [inline loafwallet/TransferAmountViewModel.swift]: Dead code. If this is needed for testing or something, can we find a different place to put this JSON?

**Review by JCHeist — APPROVED** [review]
The one request for removal of the commented out code, but approved. lgtm

File: loafwallet/TransferAmountViewModel.swift (hunk 4 above)
Comment (kcw-grunt) [reply loafwallet/TransferAmountViewModel.swift]: Ok. Thanks.

Comment (kcw-grunt) [reply loafwallet/TransferAmountViewModel.swift]: I removed that and added a `DEV://` Note

**Pull request merged**